src/extremeweatherbench/evaluate.py

```diff
@@ -2,19 +2,183 @@
+        )
+        time_subset_gridded_obs_ds = self.individual_case.perform_subsetting_procedure(
+            gridded_obs
+        )
+        # Align gridded_obs and forecast_dataset by time
+        subset_gridded_obs, forecast_ds = xr.align(
+            time_subset_gridded_obs_ds,
+            self.forecast,
+            join="inner",
+        )
+        return CaseEvaluationInput(
+            "gridded", observation=subset_gridded_obs, forecast=forecast_ds
+        )
+
+    def _subset_point_obs(self, point_obs: pd.DataFrame) -> CaseEvaluationInput:
+        """Subset the point observation dataarray for a given data variable."""
+        subset_id_point_obs = point_obs.loc[point_obs["id"] == self.individual_case.id]
+        mapped_subset_id_point_obs = subset_id_point_obs.rename(
+            columns=utils.ISD_MAPPING
+        )
+        mapped_subset_id_point_obs["longitude"] = utils.convert_longitude_to_360(
+            mapped_subset_id_point_obs["longitude"]
+        )
+        mapped_subset_id_point_obs = utils.unit_check(mapped_subset_id_point_obs)
+        mapped_subset_id_point_obs = utils.location_subset_point_obs(
+            mapped_subset_id_point_obs,
+            self.forecast["latitude"].min().values,
+            self.forecast["latitude"].max().values,
+            self.forecast["longitude"].min().values,
+            self.forecast["longitude"].max().values,
+        )
+        point_forecast_da, subset_point_obs_da = utils.align_point_obs_from_gridded(
+            self.forecast, mapped_subset_id_point_obs, utils.POINT_OBS_METADATA_VARS
+        )
+
+        point_forecast_da = point_forecast_da.groupby(
+            ["init_time", "lead_time", "latitude", "longitude"]
+        ).mean()
+        subset_point_obs_da = subset_point_obs_da.groupby(
+            ["time", "latitude", "longitude"]
+        ).first()
+        return CaseEvaluationInput(
+            "point", observation=subset_point_obs_da, forecast=point_forecast_da
+        )
+
+    def _check_forecast_data_availability(self):
+        if len(self.forecast.lead_time) == 0 or len(self.forecast.init_time) == 0:
+            raise ValueError("No forecast data available, check forecast dataset.")
+        forecast = self.individual_case._subset_valid_times(self.forecast)
+        lead_time_len = len(forecast.init_time)
+        if lead_time_len == 0:
+            logger.warning(
+                "No forecast data available for case %s, skipping",
+                self.individual_case.id,
+            )
+            return None
+        elif (
+            lead_time_len
+            < (self.individual_case.end_date - self.individual_case.start_date).days
+        ):
+            logger.warning(
+                "Fewer valid times in forecast than days in case %s, results likely unreliable",
+                self.individual_case.id,
+            )
+        logger.info("Forecast data available for case %s", self.individual_case.id)
+        return forecast
+
```

Comment:
I don't actually think any of these methods should be bundled as part of a `CaseEvaluationData`. I think they should be stand-alone methods that accept a `CaseEvaluationData` as the first argument (which is actually what you've written, if you think about it). The logic here is that your `CaseEvaluationData` is just a simple struct to keep related data tagged in one place. The helper methods use `self` to indirectly fetch data. But there's nothing special about the instance of `CaseEvaluationData` that calls them.

So just make these all stand-alone functions, and replace the argument `self` with `case_evaluation_data: CaseEvaluationData`.